### PR TITLE
parse IPv6 addresses and lua

### PR DESCRIPTION
**Pull request by cquest**

### CODE REVIEW

File: monitor/monitor.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type Line struct {
 
 func (r *RedisServer) Monitor(ctx context.Context, evt func(bool)) (chan Line, chan error) {
 	// +1619454979.381488 [1 172.29.1.2:57676] "brpop"
-	line := regexp.MustCompile(`^\+(\d+\.\d+) \[(\d+) ([\d.]+):(\d+)] "(.*?)"`)
+	line := regexp.MustCompile(`^\+(\d+\.\d+) \[(\d+) ([\d\.]+|\[[0-9a-f\:]+\]|lua):?(\d+)?\] "(.*?)"`)
```
Comment (athoune) [inline monitor/monitor.go:27]: Can you add 2 comments, one for ipv6 client, one for lua client, please.

File: monitor/monitor.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func (r *RedisServer) Monitor(ctx context.Context, evt func(bool)) (chan Line, c
 					errors <- fmt.Errorf("monitor %v %v", l, err)
 					break
 				}
-				port, err := strconv.Atoi(l[4])
+				port, err := strconv.Atoi("0"+l[4])
```
Comment (athoune) [inline monitor/monitor.go:82]: What is the meaning of a leading 0 for Atoi ?

**Review by athoune — CHANGES_REQUESTED** [review]

File: monitor/monitor.go (hunk 2 above)
Comment (cquest) [reply monitor/monitor.go:82]: When the client is lua, there is not port number, so this (dirty hack) creates a dummy 0 port number.

File: monitor/monitor.go (hunk 1 above)
Comment (cquest) [reply monitor/monitor.go:27]: IPv6 contain hexadecimal \d is not enough.

Comment (cquest) [reply monitor/monitor.go:27]: Just added IPv6 and LUA examples.

**Review by athoune — APPROVED** [review]

**Pull request merged**